### PR TITLE
feat(date-picker): add multiple prop,close #3357

### LINKED ISSUE #3357: add prop to allow selection of multiple dates on the date picker
### This function solves the problem (这个功能解决的问题)
allow to input multiple dates (not a date range!)
### Expected API (期望的 API)
https://quasar.dev/vue-components/date#multiple-selection

### PR DESCRIPTION
**Pull request by Jevon617**

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -57,6 +57,7 @@
 - `n-tree` adds `scrollTo` method, closes [#3480](https://github.com/tusen-ai/naive-ui/issues/3480).
 - `n-pagination` adds `display-order` prop, closes [#3466](https://github.com/tusen-ai/naive-ui/issues/3466).
 - `n-grid` adds `layout-shift-disabled` prop, closes [#3301](https://github.com/tusen-ai/naive-ui/issues/3301).
+- `n-date-picker` adds `multiple` prop, closes[#3357](https://github.com/tusen-ai/naive-ui/issues/3357), at the same time, this modification will affect the format of 'value' of 'type = daterange', from '[0, 0]' to '{from: 0, to: 0}'`.
```
Comment (07akioni) [inline CHANGELOG.en-US.md:60]: This won't be okay. It will break users app harshly. Why this is needed?

Comment (07akioni) [reply CHANGELOG.en-US.md:60]: I think multiple prop should only work for date-picker with `type='date'`.

Comment (Jevon617) [reply CHANGELOG.en-US.md:60]: 额, 我当初考虑的是`daterange` 时的 `value`是元组, 和 `type=date multiple` 的`value`是数组在js层面上无法区分, 需要多加一层type='daterange'来区别.

Comment (07akioni) [reply CHANGELOG.en-US.md:60]: > 额, 我当初考虑的是`daterange` 时的 `value`是元组, 和 `type=date multiple` 的`value`是数组在js层面上无法区分, 需要多加一层type='daterange'来区别.

Don't need to distinguish it by value, we should use `props.multiple` to distinguish it.

Comment (Jevon617) [reply CHANGELOG.en-US.md:60]: OK, I'll restore it.

**Pull request closed without merging**